app/controllers/documents_controller.rb

```diff
@@ -9,11 +9,24 @@ def edit
     @document = Document.find(params[:id])
   end
 
+  def show
+    @document = Document.find(params[:id])
+  end
+
   def update
     document = Document.find(params[:id])
     allowed_field_names_in_contents = document.document_type_schema.fields.map(&:id)
     document_update_params = params.require(:document).permit(:title, contents: allowed_field_names_in_contents)
     document.update_attributes(document_update_params)
-    redirect_to edit_document_path(document)
+    DocumentPublishingService.new.publish_draft(document)
+    redirect_to edit_document_path(document), notice: "Preview creation successful"
+  rescue StandardError
```

Comment:
Is this just here because this is a proof of concept? As we'd lose all exception notifications if we kept this here

Reply:
We were thinking there should be some minimal error handling in all external actions at the point we put them in. I've updated these to be more specific. In future we may want to show the specific message, but for now I've also added a logger line. Any better?